parsers/go.c

```diff
@@ -685,6 +693,15 @@ static void parseStructMembers (tokenInfo *const token, const int scope)
 			{
 				// TypeName of AnonymousField has a dot like package"."type.
 				// Throw away the package component.
+				if (!typeForAnonMember)
+					typeForAnonMember = vStringNewCopy (memberCandidate->string);
+				else
+				{
+					vStringPut (typeForAnonMember, '.');
+					vStringCat (typeForAnonMember, memberCandidate->string);
+				}
+
+
```

Comment:
Like in my previous comment, I would move this part outside of this loop and make another one just for type parsing. Then it would be good to think about how variables are named - if type parsing is detected, memberCandidate->string should be copied to something like "type" variable so it's clearer what is happening - using memberCandidate below for typeForAnonMember creation doesn't make things very obvious.